### PR TITLE
[New Rule] Persistence via Folder Action Script

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/684

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/persistence_folder_action_scripts_runtime.toml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+[metadata]
+creation_date = "2020/12/07"
+maturity = "production"
+updated_date = "2020/12/07"
+
+[rule]
+author = ["Elastic"]
+description = """
+A Folder Action script is executed when the folder to which it is attached has items added or removed, or when its
+window is opened, closed, moved, or resized. Adversaries may abused this feature to persist a malicious script.
```
Comment (bm11100) [inline rules/macos/persistence_folder_action_scripts_runtime.toml:10]:
```suggestion
A Folder Action script is executed when the folder to which it is attached has items added or removed, or when its
window is opened, closed, moved, or resized. Adversaries may abuse this feature to establish persistence by utilizing a malicious script.
```

Comment (bm11100) [reply rules/macos/persistence_folder_action_scripts_runtime.toml:10]: may need a re-lint

**Review by bm11100 — APPROVED** [review]

File: rules/macos/persistence_folder_action_scripts_runtime.toml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+risk_score = 47
+rule_id = "c292fa52-4115-408a-b897-e14f684b3cb7"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Execution", "Persistence"]
+type = "eql"
+
+query = '''
+sequence by host.id with maxspan = 5s
+ [process where event.type in ("start", "process_started", "info") and process.name : "com.apple.foundation.UserScriptService"] by process.pid
+ [process where event.type in ("start", "process_started") and process.name : ("osascript", "sh")] by process.ppid
```
Comment (brokensound77) [inline rules/macos/persistence_folder_action_scripts_runtime.toml:27]:
```suggestion
sequence by host.id with maxspan=5s
 [process where event.type in ("start", "process_started", "info") and process.name == "com.apple.foundation.UserScriptService"] by process.entity_id
 [process where event.type in ("start", "process_started") and process.name in ("osascript", "sh")] by process.parent.entity_id
```

**Review by brokensound77 — APPROVED** [review]
Tweak to query then LGTM

File: rules/macos/persistence_folder_action_scripts_runtime.toml (hunk 2 above)
Comment (Samirbous) [reply rules/macos/persistence_folder_action_scripts_runtime.toml:27]: replaced back entity_id with pid or ppid (not populated in auditbeat)

**Pull request merged**